plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -484,27 +537,99 @@ public void truncateTable(ConnectorSession session, ConnectorTableHandle tableHa
     @Override
     public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle, List<ColumnHandle> columns, RetryMode retryMode)
     {
-        if (retryMode != RetryMode.NO_RETRIES) {
-            throw new TrinoException(NOT_SUPPORTED, "This connector does not support query retries");
-        }
         BigQueryTableHandle table = (BigQueryTableHandle) tableHandle;
         if (isWildcardTable(TableDefinition.Type.valueOf(table.asPlainTable().getType()), table.asPlainTable().getRemoteTableName().getTableName())) {
             throw new TrinoException(BIGQUERY_UNSUPPORTED_OPERATION, "This connector does not support inserting into wildcard tables");
         }
         ImmutableList.Builder<String> columnNames = ImmutableList.builderWithExpectedSize(columns.size());
         ImmutableList.Builder<Type> columnTypes = ImmutableList.builderWithExpectedSize(columns.size());
+        ImmutableList.Builder<Field> tempFields = ImmutableList.builderWithExpectedSize(columns.size() + 1);
+
         for (ColumnHandle columnHandle : columns) {
             BigQueryColumnHandle column = (BigQueryColumnHandle) columnHandle;
+            tempFields.add(toField(column.getName(), column.getTrinoType(), column.getColumnMetadata().getComment()));
             columnNames.add(column.getName());
             columnTypes.add(column.getTrinoType());
         }
-        return new BigQueryInsertTableHandle(table.asPlainTable().getRemoteTableName(), columnNames.build(), columnTypes.build());
+        ColumnMetadata pageSinkIdColumn = getPageSinkIdColumn(columnNames.build());
+        tempFields.add(toField(pageSinkIdColumn.getName(), pageSinkIdColumn.getType(), pageSinkIdColumn.getComment()));
+
+        BigQueryClient client = bigQueryClientFactory.create(session);
+        String projectId = ((BigQueryNamedRelationHandle) table.getRelationHandle()).getRemoteTableName().getProjectId();
+        String schemaName = ((BigQueryNamedRelationHandle) table.getRelationHandle()).getSchemaTableName().getSchemaName();
```

Comment:
Why `projectId` is retrieved via `getRemoteTableName` and `schemaName` is retrieved via `getSchemaTableName` (why not `RemoteTableName#getDatasetName`)?

Reply:
i think this was a mistake